src/script/cpp_api/s_security.cpp

```diff
@@ -963,3 +979,34 @@ int ScriptApiSecurity::sl_os_setlocale(lua_State *L)
 	lua_call(L, cat ? 2 : 1, 1);
 	return 1;
 }
+
+
+int ScriptApiSecurity::sl_debug_getinfo(lua_State *L)
+{
+	// signature: [thread,] function [, what]
+	const bool thread = lua_isthread(L, 1);
+	const int funidx = thread ? 2 : 1;
+	const bool fun = lua_isfunction(L, funidx);
+	const bool what = lua_gettop(L) > funidx;
+
+	push_original(L, "debug", "getinfo");
+	if (thread)
+		lua_pushvalue(L, 1);
+	if (fun) {
+		lua_pushvalue(L, funidx);
+	} else {
+		// +1 to account for wrapper (unless coroutine)
+		lua_pushinteger(L, luaL_checkinteger(L, funidx) + (thread ? 0 : 1));
```

Comment:
Fun can be none, and then at funidx there will be debug.getinfo. Please call `luaL_checkinteger` earlier for proper argument checking.